### DISCUSSION TITLE
help with Decompiler.exe

### DISCUSSION THREAD

**Discussion by staaandup** [SteamDatabase/ValveResourceFormat#416] [General]
Does anyone know a magic command i can add to this batch file to speed up the extraction? i need to extract thousands of specific files, and this method takes far too long.

```
@echo off
Decompiler.exe -i "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" -d -f "particles\addons_gameplay\fire_torch_glow_shadowed.vpcf_c" -o "C:\Output"
Decompiler.exe -i "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" -d -f "particles\addons_gameplay\fountain_tintable_ambient.vpcf_c" -o "C:\Output"
Decompiler.exe -i "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" -d -f "particles\addons_gameplay\fountain_tintable_radiance.vpcf_c" -o "C:\Output"
Decompiler.exe -i "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" -d -f "particles\addons_gameplay\fountain_tintable_waves.vpcf_c" -o "C:\Output"
...And more
```

**Comment by xPaw** [comment]
There is `--threads` option if you export/decompile an entire folder.

**Reply by staaandup** [reply]
@xPaw Thank you for your response. However, I only want to extract specific files from each folder, not the entire folder and all of the files contained within it. So I don't think `--threads` will be useful in this case?

**Reply by kristiker** [reply]
You are heading on a very strange direction. Why need 1000s of specific files decompiled ectracted? Go with the simple route: do them all and pick those that you want.

**Reply by staaandup** [reply]
![2022-04-11 17_25_01-Fade Lens](https://user-images.githubusercontent.com/50317210/162788175-5fd4ef26-d573-4de9-87bd-3bbc97389e9b.png)
I simply can't. i'd rather extract specific files than the entire folder because the entire folder contains about 60k files and 3k subfolders and it takes me 20 to 30 minutes to extract everything on my hdd. even the subfolders are large and full of files that i don't need, so i need a command that extracts only the files that i need and does it quickly (in total they are about 4k).

**Comment by Panda19213** [comment]
Здраствуйте, I am a medical student and I wanted to pull out one model from Dota2 and remake it for myself, I do not know how programming works at all. I found a discussion in Steame where you left a link to pull out the vmdl model, how to use what's inside the link? Maybe there are some videos that teach?
